### PR TITLE
Enhance Api Platform Profiler

### PR DESCRIPTION
**Pull request by antograssiot**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

This PR aims to improve the Datacollector by : 

- Use the `profiler_dump()` function for a cleaner rendering
- Format the keys of the arrays and limit their size in the array
- Highlight the Resource class and move the Acceptable content type at the bottom
- Presents the attributes on 2 columns only
- Highlight the actual operation
- Add the resource class display from the toolbar

@jdeniau I hope this is going in the direction you initially had in mind, and that the change from 3 to 2 columns is ok, IMO mot of the time the really interesting information is/was not the method nor route name only.

I remove the request method from the collector has it was not used in the template and this information is already present in other panels...

before: 
screenshot-2018-4-8 symfony profiler 1

after:
screenshot-2018-4-8 symfony profiler

capture d ecran 2018-04-08 a 08 45 52

If people like it, I'd like then to improve the filter parts by injecting the filter locator into the dataCollector, that would allow:
 - to expose at the Filter Class at least (but maybe also attributes etc)
 - to highlight in read the filters declared that doesn't match a real filter service that are "silently ignored" today IIRC

Another idea that I have would be to display the actual result number near by the toolbar icon.
The idea would be to check if the result is a Paginator and get the total items number out of it or display 1 on an item operation. (It is sometime not quick to get this information as it stands in the bottom of the response in JSON-LD) at least

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig [hunk 1]
```diff
@@ -1,33 +1,9 @@
...
-            >
-                Show
-            </a>
-        </td>
-    </tr>
-    <tr id="api-platform-line-{{ itemOrCollection }}-{{ key }}">
-        <td colspan="3">
-            {{- dump(operation) -}}
-        </td>
+        <th scope="row" {{ key == actualOperationName ? 'class="status-success"' : '' }}>{{ key }}</th>
```
Comment (sroze) [inline src/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig]: My HTML purism would say:
```patch
- <th scope="row" {{ key == actualOperationName ? 'class="status-success"' : '' }}>{{ key }}</th>
+ <th scope="row"{{ key == actualOperationName ? ' class="status-success"' : '' }}>{{ key }}</th>
```

Comment (dunglas) [reply src/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig]: I would even use:

`<th scope="row"{% if key == actualOperationName %} class="status-success"{% endif %}>{{ key }}</th>`

File: src/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig [hunk 2]
```diff
@@ -1,33 +1,9 @@
...
+        <td {{ key == actualOperationName ? 'class="status-success"' : '' }}>{{- profiler_dump(operation, 1) -}}</td>
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/Resources/views/DataCollector/request.html.twig]: Same here btw:

```patch
-<td {{ key == actualOperationName ? 'class="status-success"' : '' }}>
+<td{% if key == actualOperationName %} class="status-success"{% endif %}>
```

**Comment by antograssiot** [comment]
glad you liked it.
updated as requested

**Comment by antograssiot** [comment]
rebased, anything else to be done on this one @dunglas ?

**Comment by Simperfit** [comment]
Since @dunglas's comments have been changed accordingly and that we have two approuve I'm merging this one.

Thanks @antograssiot.

**Comment by sroze** [comment]
It introduced a bug 🙃 See #1846.

**Comment by sroze** [comment]
Great move 😉

**Pull request merged**